packages/cli/src/cli/overview/action.create.overview.ts

```diff
@@ -0,0 +1,176 @@
+    for (const file of files) {
+      const name = file.name;
+      const [z, x, y] = path.basename(name).replace('.tiff', '').split('-').map(Number);
+      let qk = QuadKey.fromTile({ x, y, z });
+      this.addChildren(qk, maxZoom);
+      while (qk.length > 0) {
+        if (this.tiles.has(qk)) break;
+        this.tiles.add(qk);
+        qk = QuadKey.parent(qk);
+      }
+    }
+  }
+
+  addChildren(qk: string, maxZoom: number): void {
+    if (qk.length >= maxZoom) return;
+    for (const child of QuadKey.children(qk)) {
+      this.tiles.add(child);
+      if (child.length < maxZoom) this.addChildren(child, maxZoom);
+    }
+  }
+
+  async prepareSourceFiles(sources: ChunkSource[]): Promise<TileMatrixSet> {
+    let tileMatrix;
+    for (const source of sources) {
+      const tiff = await CogTiff.create(source);
+      await tiff.getImage(0).loadGeoTiffTags();
+      if (tileMatrix == null) tileMatrix = TileMatrixSets.tryGet(tiff.getImage(0).epsg);
+      const imgBounds = Bounds.fromBbox(tiff.getImage(0).bbox);
+      this.files.push({
+        name: tiff.source.uri,
+        ...imgBounds,
+      });
+    }
+    if (tileMatrix == null) throw new Error('Unable to find the imagery tileMatrix');
+    return tileMatrix;
+  }
+
+  async generateTiles(path: string, tileMatrix: TileMatrixSet): Promise<void> {
+    const promises = [];
+    let currentTiles = Array.from(this.tiles);
+    while (currentTiles.length > 0) {
+      const todo = currentTiles.slice(0, WorkerTaskSize);
+      currentTiles = currentTiles.slice(WorkerTaskSize);
+      const jobTiles: JobTiles = { path, files: this.files, tileMatrix: tileMatrix.identifier, tiles: todo };
+      promises.push(pool.run('tile', jobTiles));
+    }
+
+    await Promise.all(promises);
+    await pool.close();
+  }
+
+  async createTar(path: string, logger: LogType): Promise<void> {
+    const tarFile = 'overview.co.tar';
+    const tarIndex = 'overview.tar.index';
+    const tarFilePath = fsa.join(path, tarFile);
+    const tarIndexPath = fsa.join(path, tarIndex);
+
+    // Create tar file
+    const tiles = await fsa.toArray(fsa.list(`${path}/tiles/`));
+    const cwd = fsa.join(process.cwd(), path);
+    const paths = tiles.map((c) => c.replace(`${cwd}/`, ''));
+    await tar.c({ file: tarFilePath, C: cwd }, paths);
+
+    // Creating tar index
+    const fd = await fs.open(tarFilePath, 'r');
+    const opts: CotarIndexOptions = { packingFactor: 1.25, maxSearch: 50 }; // Default package rule.
+    const index = await CotarIndexBuilder.create(fd, opts);
+    const indexBinary = await CotarIndexBinary.create(new SourceMemory('index', index.buffer));
+    await TarReader.validate(fd, indexBinary);
+    await fs.writeFile(tarIndexPath, index.buffer);
+
+    // Copy the output into s3 location
+    const output = this.output.value;
+    if (output) {
+      logger.info({ output }, 'CreateOverview: UploadOutput');
+      await fsa.write(fsa.join(output, tarFile), await fsa.read(tarFilePath));
```

Comment:
this should be one file, you shouldnt write the index seperatly.

Reply:
This is just additional `--output` parameter to copy the tar file and index into s3.